athena-db2/src/main/java/com/amazonaws/athena/connectors/db2/Db2MetadataHandler.java

```diff
@@ -0,0 +1,503 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Db2MetadataHandler extends JdbcMetadataHandler
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(Db2MetadataHandler.class);
+    static final String PARTITION_NUMBER = "PARTITION_NUMBER";
+    static final String PARTITIONING_COLUMN = "PARTITIONING_COLUMN";
+    /**
+     * DB2 has max number of partition 32,000
+     */
+    private static final int MAX_SPLITS_PER_REQUEST = 32000;
+
+    /**
+     * Instantiates handler to be used by Lambda function directly.
+     * <p>
+     * Recommend using {@link Db2MuxCompositeHandler} instead.
+     */
+    public Db2MetadataHandler()
+    {
+        this(JDBCUtil.getSingleDatabaseConfigFromEnv(Db2Constants.NAME));
+    }
+
+    /**
+     * TO be used by Mux.
+     *
+     * @param databaseConnectionConfig
+     */
+    public Db2MetadataHandler(final DatabaseConnectionConfig databaseConnectionConfig)
+    {
+        this(databaseConnectionConfig, new GenericJdbcConnectionFactory(databaseConnectionConfig, null, new DatabaseConnectionInfo(Db2Constants.DRIVER_CLASS, Db2Constants.DEFAULT_PORT)));
+    }
+
+    /**
+     * To be used by Mux.
+     *
+     * @param databaseConnectionConfig
+     * @param jdbcConnectionFactory
+     */
+    public Db2MetadataHandler(final DatabaseConnectionConfig databaseConnectionConfig, final JdbcConnectionFactory jdbcConnectionFactory)
+    {
+        super(databaseConnectionConfig, jdbcConnectionFactory);
+    }
+
+    @VisibleForTesting
+    protected Db2MetadataHandler(final DatabaseConnectionConfig databaseConnectionConfig, final AWSSecretsManager secretsManager,
+                                 AmazonAthena athena, final JdbcConnectionFactory jdbcConnectionFactory)
+    {
+        super(databaseConnectionConfig, secretsManager, athena, jdbcConnectionFactory);
+    }
+
+    /**
+     * Overridden this method to fetch only user defined schema(s) in Athena Data window.
+     *
+     * @param blockAllocator
+     * @param listSchemasRequest
+     * @return
+     */
+    @Override
+    public ListSchemasResponse doListSchemaNames(final BlockAllocator blockAllocator, final ListSchemasRequest listSchemasRequest)
+    {
+        try (Connection connection = getJdbcConnectionFactory().getConnection(getCredentialProvider())) {
+            LOGGER.info("{}: List schema names for Catalog {}", listSchemasRequest.getQueryId(), listSchemasRequest.getCatalogName());
+            return new ListSchemasResponse(listSchemasRequest.getCatalogName(), getSchemaList(connection, Db2Constants.QRY_TO_LIST_SCHEMAS));
+        }
+        catch (SQLException sqlException) {
+            throw new RuntimeException(sqlException.getErrorCode() + ": " + sqlException.getMessage());
+        }
```

Comment:
this code spinet method signature (and its parent) does not have throw Exception, therefor we are bound to write try/catch block and rethrow exception.  We have changed our private method signature and added try/catch and rethrow exception.

Reply:
Ok I see that the JdbcMetadataHandler is catching just to rethrow as well. I will talk with the team and fix that underlying class to not do that, so that you also don't have to catch and rethrow anymore.

Reply:
Hey Feroz, I've updated our codebase to eliminate this behavior now. You should rebase and then you can eliminate these catch and rethrows now: https://github.com/awslabs/aws-athena-query-federation/commit/2278a82e89f0bfc946107e3badf89c926910bd6f